core/blockchain_af.go

```diff
@@ -65,6 +66,9 @@ func (bc *BlockChain) EnableArtificialFinality(enable bool, logValues ...interfa
 // finality feature setting.
 // This status is agnostic of feature activation by chain configuration.
 func (bc *BlockChain) IsArtificialFinalityEnabled() bool {
+	if bc.Config().GetConsensusEngineType() != ctypes.ConsensusEngineT_Ethash {
+		return false
+	}
```

Comment:
This should be a practical redundancy since AF is only reasonably applied to PoW consensus engines, but this enforces that. Questionably, this could cause silently unexpected operation if a user _does_ try to use an AF feature with a non-PoW protocol, since they won't receive an error or refusal on startup (implying that this is a logically half-assed change).